runtime/src/accounts_db.rs

```diff
@@ -2447,11 +2448,33 @@ impl AccountsDb {
         });
         let dirty_stores_len = dirty_stores.len();
         let pubkeys = DashSet::new();
-        timings.oldest_dirty_slot = max_slot.saturating_add(1);
-        for (slot, store) in dirty_stores {
-            timings.oldest_dirty_slot = std::cmp::min(timings.oldest_dirty_slot, slot);
-            store.accounts.account_iter().for_each(|account| {
-                pubkeys.insert(account.meta.pubkey);
+        let mut dirty_store_routine = || {
+            let chunk_size = 1.max(dirty_stores_len.saturating_div(rayon::current_num_threads()));
+            let oldest_dirty_slots: Vec<u64> = dirty_stores
+                .par_chunks(chunk_size)
+                .map(|dirty_store_chunk| {
+                    let mut oldest_dirty_slot = max_slot.saturating_add(1);
+                    dirty_store_chunk.iter().for_each(|(slot, store)| {
```

Comment:
yes. This could be a page fault to iterate the append vec on disk. Makes sense we may want to request to page in multiple files in parallel as opposed to sequentially.